pkg/app/ops/insightcollector/collector.go

```diff
@@ -160,12 +161,28 @@ func (c *InsightCollector) ProcessNewlyCompletedDeployments(ctx context.Context)
 	return nil
 }
 
-func (c *InsightCollector) ProcessApplications(_ context.Context) error {
+func (c *InsightCollector) ProcessApplications(ctx context.Context) error {
 	c.logger.Info("will retrieve all applications to build insight data")
 	if len(c.newlyCreatedDeploymentsHandlers) == 0 {
 		c.logger.Info("skip building insight data for applications because there is no configured handlers")
 		return nil
 	}
 
-	return errors.New("not implemented yet")
+	now := time.Now()
+	targetDate := time.Date(now.Year(), now.Month(), now.Day(), 0, 0, 0, 0, time.UTC)
```

Comment:
I think we can use `now` directly. This `ProcessApplication` can be called more frequently. e.g for each 2 hours.